extensions/answer_summarizers/models.py

```diff
@@ -90,6 +96,52 @@ def _get_top_answers_by_frequency(answers, limit=None):
     ])
 
 
+def _get_top_unresolved_answers_by_frequency(
+        answers_with_classification, limit=None):
+    hashable_answers_with_classification = [{
+        'hashable_answer': _HashableAnswer(ans['answer']),
+        'classification_categorization': ans['classification_categorization']
+    } for ans in answers_with_classification]
+
+    classification_results_dict = {}
+    for ans in hashable_answers_with_classification:
+        frequency = 0
+        if ans['hashable_answer'] in classification_results_dict:
```

Comment:
I think I'm a little confused about what's going on here. It seems to be implicitly assumed that a given hashable answer in hashable_answers_with_classification will always have the same classification_categorization. Why is this true?

Reply:
No, that is not true. A given answer may be repeated many times in hashable_answers_with_classification with different classification categorization. What I am doing is that I am keeping track of their latest classification result. So only those answers whose most recent classification is one of the unresolved ones are classified as unresolved answers.

Reply:
Ah, I see, thanks for explaining. Could you add a comment here, and also a docstring to the function explaining that the original list is sorted in time order? I did not realize that from reading the code.

Also, how do we know that the original list is sorted in time order -- what guarantees that?

Reply:
Because answers are stored in the original list as they come and so it is time-sorted implicitly. If we change that then we will have to change this algorithm, too.

Reply:
OK. I think this must be stated somewhere, e.g. in the contract part of the docstring on line 309. Something like: "The answers in this dict must be sorted in chronological order." And maybe in whatever feeds into that as well, all the way down to the datastore model.

This will help ensure that the invariant doesn't change if we ever do, e.g., a migration. Otherwise the fact that they currently happen to be stored in time order is implicit, undocumented, and may break in the future.

Reply:
OK, Added the comment. Should I also add a comment in StatsModel?

Reply:
Yep, all the way to the source. Thanks!

Reply:
Done